modules/ROOT/examples/java/src/com/couchbase/code_snippets/Examples.java

```diff
@@ -1766,3 +1768,35 @@ public void log(@NonNull LogLevel level, @NonNull LogDomain domain, @NonNull Str
     }
 }
 // end::custom-logging[]
+
+
+
+
+// tag::ziputils-unzip[]
+public class ZipUtils {
+    public static void unzip(InputStream in, File destination) throws IOException {
+        byte[] buffer = new byte[1024];
+        ZipInputStream zis = new ZipInputStream(in);
+        ZipEntry ze = zis.getNextEntry();
+        while (ze != null) {
+            String fileName = ze.getName();
+            File newFile = new File(destination, fileName);
+            if (ze.isDirectory()) {
+                newFile.mkdirs();
+            } else {
+                new File(newFile.getParent()).mkdirs();
+                FileOutputStream fos = new FileOutputStream(newFile);
+                int len;
+                while ((len = zis.read(buffer)) > 0) {
+                    fos.write(buffer, 0, len);
+                }
+                fos.close();
+            }
+            ze = zis.getNextEntry();
+        }
```

Comment:
I think that these calls to `close` need to be in a finally block... or, better yet, in a `try (resource) {`